### PR TITLE
I1039 write to hdfs

### LINKED ISSUE #1039: Provide API to write to HDFS
Need to be usable by Python, so may involve conversion from Python stream to Java stream.

### PR DESCRIPTION
**Pull request by Kai-Chen**
Wrapper for Python file object to be used in Java I/O.

Closes #1039

### CODE REVIEW

File: src/main/scala/org/tresamigos/smv/SmvHDFS.scala [hunk 1]
```diff
@@ -120,3 +158,21 @@ private[smv] object SmvHDFS {
       r = deleteFile(filename)
     } yield (filename, r)
 }
+
+class InputStreamAdapter(in: InputStream) extends IAnyInputStream {
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/SmvHDFS.scala]: comment to explain what this is.

File: src/main/scala/org/tresamigos/smv/SmvHDFS.scala [hunk 2]
```diff
@@ -51,16 +69,36 @@ private[smv] object SmvHDFS {
...
+    val from = InputStreamAdapter(contents)
+    try {
+      writeToFile(from, fileName)
+    } finally {
+      from.close()
+    }
+  }
+
+  @scala.annotation.tailrec
+  def _copy(in: IAnyInputStream, out: OutputStream, chunksize: Int, buf: Array[Byte]): Unit =
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/SmvHDFS.scala]: use imperative `while` loop for us non lispish people :-)

File: src/main/scala/org/tresamigos/smv/SmvHDFS.scala [hunk 3]
```diff
@@ -120,3 +158,21 @@ private[smv] object SmvHDFS {
       r = deleteFile(filename)
     } yield (filename, r)
 }
+
+class InputStreamAdapter(in: InputStream) extends IAnyInputStream {
+  @Override def read(max: Int) = {
+    val buf = new Array[Byte](max)
+    in.read(buf) match {
+      case -1 => Array.empty
+      case 0 => read(max)
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/SmvHDFS.scala]: potential stack overflow.

**Comment by AliTajeldin** [comment]
@Kai-Chen did you test it against 3.5?
Also needs a merge with master

**Comment by Kai-Chen** [comment]
Yes, test passes with 3.5

**Pull request merged**